### PR TITLE
visualziation selection for query builder

### PR DESCRIPTION
**Pull request by eric-j**
https://new-relic.atlassian.net/browse/NR-218056

### CODE REVIEW

File: shared/agent/src/providers/newrelic/nrql/nrqlProvider.ts [hunk 1]
```diff
@@ -302,16 +307,22 @@ export class NrNRQLProvider {
...
+					Object.keys(value).length === 1 && !Array.isArray(value[Object.keys(value)[0]])
+						? "billboard"
+						: "json";
+				return {
+					selected: selectedValue,
+					enabled: selectedValue === "billboard" ? ALL_RESULT_TYPES : ["json"],
+				};
 			} else {
-				return "billboard";
+				return { selected: "billboard", enabled: ALL_RESULT_TYPES };
```
Comment (bcanzanella) [inline shared/agent/src/providers/newrelic/nrql/nrqlProvider.ts:325]: technically this works, though not terribly useful. on NR1 billboard's only other alternatives are ["json"] (and bullet, which we dont support)

File: shared/ui/Stream/Dropdown.tsx [hunk 2]
```diff
@@ -71,15 +71,22 @@ export const Dropdown = (props: Props) => {
 		props.selectedValue || ""
 	);
 
+	useEffect(() => {
```
Comment (bcanzanella) [inline shared/ui/Stream/Dropdown.tsx:74]: there any risk[s] in modifying this component? i ask because it's used in a lot of places.

**Comment by bcanzanella** [comment]
couple ❓ s otherwise LGTM

File: shared/agent/src/providers/newrelic/nrql/nrqlProvider.ts (hunk 1 above)
Comment (eric-j) [reply shared/agent/src/providers/newrelic/nrql/nrqlProvider.ts:325]: good point, ill change

File: shared/ui/Stream/Dropdown.tsx (hunk 2 above)
Comment (eric-j) [reply shared/ui/Stream/Dropdown.tsx:74]: Low risk, and this actually fixes some wonky behavior where its currently implemented as well.

Comment (bcanzanella) [reply shared/ui/Stream/Dropdown.tsx:74]: ok sweet

**Review by bcanzanella — APPROVED** [review]

**Pull request merged**